### DISCUSSION TITLE
How to debug SMTP issues in docker?

### DISCUSSION THREAD

**Discussion by ccoffee** [requarks/wiki#3909] [Help / Questions]
Been scraping docs and such for a while. Cannot figure out how to debug issues I am having with SMTP. 

I am having issues connecting to Google smtp and get error and I cannot find any smtp related logs in the docker container or otherwise. I don't see any smtp or logging parameters in the config file and all the options I have tried have not worked.
The most basic docker install on Digital Ocean. The same options for SMTP setup in Outlook and via a Node script work fine.

How can I see the SMTP request payloads it is making and the full response from Google's SMTP server?
My current guess is it is not using the correct hostname when sending the request to Google's SMTP server and I can't find any information to configure that in Wiki.js.

**Comment by vwlau** [comment]
I did `docker exec -it wiki bash` and dug around the docker container a bit and was unable to find where the logs are. 

However, I was able to set up gmail smtp. I faced similar issues setting up outgoing emails for another docker container and picked up some tricks from getting that to work. 

If you don't have 2-factor sign on enabled for the gmail account you are sending from, try going to your google account settings -> security and enabling "Less Secure Apps." 

If you do have 2-factor sign on enabled, you need to create an app password in place of your google/gmail account password. Account settings -> Security -> App Passwords. Under "Select the app and device you want to generate the app password for." Pick "Mail" for the "Select App" dropdown and "Other (Custom Name)" for the "Select Device" dropdown. Type in whatever name you want (I put "wiki js mail") and hit generate. You'll get a popup with for I think a 16 character password in a highlighted box.  Copy that into the password field of the mail configuration instead of your normal password for the gmail account you are using. 

General settings:

Host: smtp.gmail.com
Port: 465
Secure TLS: enabled 
Verify SSL Certifcate: enabled (I have certbot doing my ssl certificates with an nginx reverse proxy.) 

I believe that sender email and username need to match but I didn't check for sure. 

I am not using DKIM. 

I know I didn't really answer your question, but I hope that helps anyway!

**Reply by dcpc007** [reply]
Where did you find the logs ?
i the wiki container, i didn't find any *log* file.
Nor recent modified files (except the /data/repo )
